### DISCUSSION TITLE
Expected 0 arguments, got 1 in match?

### DISCUSSION THREAD

**Discussion by sgnix** [gvergnaud/ts-pattern#210] [Q&A] [answered]
Playing around with ts-match, I get a type error in the `match` function. It appears that the codesandbox examples have the same error, for example https://codesandbox.io/p/sandbox/ts-pattern-api-examples-bdy5p2?file=%2Fsrc%2Fexamples%2Fbasic.tsx%3A24%2C14

![Screen Shot 2024-01-05 at 12 09 37 PM](https://github.com/gvergnaud/ts-pattern/assets/606236/f27a727d-ca14-450d-890a-b59e88c32b51)

Am I missing anything?

**Comment by gvergnaud** [comment]
TS-Pattern v5+ needs TypeScript v5 or higher and your project must be in strict mode. Can you confirm that your project has an expected config?

**Comment by sgnix** [comment]
Yes, I can confirm that the project is using TypeScript version 5.3.3 and strict mode. I still get that type error. The official ts-match examples in [codesandbox](https://codesandbox.io/p/sandbox/ts-pattern-api-examples-bdy5p2?file=%2Fsrc%2Fexamples%2Fbasic.tsx%3A14%2C11) also get the same type error.

**Comment by gvergnaud** [comment]
I'm pretty sure this is because codesandbox doesn't use TS v5+, const type parameters doesn't seem supported: [See the syntax error here](https://codesandbox.io/p/sandbox/ts-pattern-api-examples-bdy5p2?file=%2Fnode_modules%2F.store%2Fts-pattern%405.0.6%2Fnode_modules%2Fts-pattern%2Fdist%2Fpatterns.d.ts%3A92%2C40). 

All tests pass with TS v5.3.3 locally

**Comment by sgnix** [comment]
Codesandbox uses TS 5.3.3. You can verify by inspecting the `package.json` file. 
Additionally, using `ts-match` locally using a fresh install (TS 5.3.3) and strict mode, still produces the same type error.
I can't be the only one seeing this!
Anyone else?

**Comment by darrylnoakes** [comment]
I think the issue is the CodeSandbox TS integration, not TS itself.

Everything _runs_ fine.

To corroborate what @gvergnaud said about `const` type parameters:
In `match.d.ts`:

![image](https://github.com/gvergnaud/ts-pattern/assets/67335139/e99a0f16-fddd-46ae-b51e-79a3414ad953)

These errors cause the declaration to fail, and `match` to be typed as `(alias) match(): any`.

**Comment by gvergnaud** [comment, marked as answer]
I switched the sandbox to a ["devbox"](https://codesandbox.io/docs/learn/devboxes/overview) and it seems like the editor is now using the version specified in the package.json and the issue is gone. 

@sgnix your editor is likely not using the latest typescript version, you should be able to fix that by clicking on "Select TypeScript Version" in VSCode's CMD+P menu.

**Comment by sgnix** [comment]
All good now. It looks like my editor's LSP plugin had an outdated TS version. Thanks for your time!
